packages/loader/src/MaterialLoader.ts

```diff
@@ -32,70 +32,74 @@ class MaterialLoader extends Loader<Material> {
       })
         .then((materialSchema: IMaterialSchema) => {
           const engine = resourceManager.engine;
-          const { name, shader, shaderData, macros, renderState } = materialSchema;
-          const material = new Material(engine, Shader.find(shader));
-          material.name = name;
+          const { name, shaderData, macros, renderState, shaderRef } = materialSchema;
 
-          const texturePromises = new Array<Promise<Texture2D>>();
-          const materialShaderData = material.shaderData;
-          for (let key in shaderData) {
-            const { type, value } = shaderData[key];
+          // @ts-ignore
+          resourceManager.getResourceByRef<Shader>(<IAssetRef>shaderRef).then((shader) => {
+            const material = new Material(engine, shader);
```

Comment:
force use shader?pls compatible with common materials and previous materials

Reply:
done